qutebrowser/misc/notree.py

```diff
@@ -0,0 +1,240 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2015-2018 Alexander Cogneau (acogneau) <alexander.cogneau@gmail.com>
+# Copyright 2015-2018 Florian Bruhin (The-Compiler) <me@the-compiler.org>
+
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tree library for tree-tabs.
+
+The fundamental unit is the Node class.
+
+Create a tree with with Node(value, parent):
+root = Node('foo')
+child = Node('bar', root)
+child2 = Node('baz', root)
+child3 = Node('lorem', child)
+
+You can also assign parent after instantiation, or even reassign it:
+child4 = Node('ipsum')
+child4.parent = root
+
+Assign children:
+child.children = []
+child2.children = [child4, child3]
+child3.parent
+> Node('foo/bar/baz')
+
+Render a tree with render_tree(root_node):
+render_tree(root)
+
+> ('', 'foo')
+> ('├─', 'bar')
+> ('│ ├─', 'lorem')
+> ('│ └─', 'ipsum')
+> ('└─', 'baz')
+"""
+import enum
+
+# For Node.render
+corner = '└─'
```

Comment:
Can these be all-caps to signify that they're "constant"

Reply:
I made them all-caps for now. However, in future I think these characters could deserve their own settings entry so users can customize it (e.g. swap them with `╟, ╚, ║`), although that could be overkill in terms of customization. What do you think?

Reply:
If we don't change these, people can (unofficially) change them in a method similar to https://github.com/qutebrowser/qutebrowser/pull/4089 .

I can see some fonts not handling those characters correctly, so I could imagine people wanting to switch them out for something like `|`, `=`, `>`. I'm not sure what a clean way to expose this would be, it does seem overkill for a setting...

Reply:
Okay then. #1578 would make it really straightforward to change it, but I don't think it's been implemented/merget yet. Until then, #4089 will still be a decent workaround.